### PR TITLE
feat(cli/support): confirm before creating bundle

### PR DESCRIPTION
**Pull request by johnstcn**
Last must-haves of https://github.com/coder/coder/issues/12163

Forces user to confirm before creating a support bundle.
Also adds contextual information to the bundle under `cli_logs.txt`.
~Finally, un-hides the `support` command`.~ EDIT: will un-hide closer to release.

### CODE REVIEW

File: cli/support.go [hunk 1]
```diff
@@ -109,13 +169,26 @@ func (r *RootCmd) supportBundle() *serpent.Command {
 		},
 	}
 	cmd.Options = serpent.OptionSet{
+		{
+			Flag:          "confirm",
+			FlagShorthand: "y",
+			Env:           "CODER_SUPPORT_BUNDLE_CONFIRM",
+			Description:   "By setting this to true, you confirm that you will treat the resulting support bundle as if it contained sensitive information.",
```
Comment (mafredri) [inline cli/support.go:176]:
```suggestion
			Description:   "By setting this, you confirm that you will treat the resulting support bundle as if it contained sensitive information.",
```
As a flag, you don't necessarily "set to true", you just provide it. Up to you if you want to change this.

**Review by mafredri — APPROVED** [review]

File: cli/support.go [hunk 2]
```diff
@@ -109,13 +169,26 @@ func (r *RootCmd) supportBundle() *serpent.Command {
...
+			Value:         serpent.BoolOf(&confirm),
+		},
 		{
 			Flag:          "output",
 			FlagShorthand: "o",
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/support.go:181]: Good catch 👍

File: cli/support.go [hunk 3]
```diff
@@ -109,13 +169,26 @@ func (r *RootCmd) supportBundle() *serpent.Command {
 		},
 	}
 	cmd.Options = serpent.OptionSet{
+		{
+			Flag:          "confirm",
+			FlagShorthand: "y",
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/support.go:174]: 👍

File: cli/support.go [hunk 4]
```diff
@@ -48,14 +52,52 @@ func (r *RootCmd) supportBundle() *serpent.Command {
...
 		Handler: func(inv *serpent.Invocation) error {
-			var (
-				log = slog.Make(sloghuman.Sink(inv.Stderr)).
-					Leveled(slog.LevelDebug)
-				deps = support.Deps{
-					Client: client,
-					Log:    log,
-				}
+			var cliLogBuf bytes.Buffer
+			cliLogW := sloghuman.Sink(&cliLogBuf)
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/support.go:56]: It looks like it does:

```
[cdr.dev/slog/sloggers/sloghuman/sloghuman.go]
func Sink(w io.Writer) slog.Sink {
	return &humanSink{
		w:  syncwriter.New(w),
		w2: w,
	}
}

[cdr.dev/slog/internal/syncwriter/syncwriter.go]
type Writer struct {
	mu sync.Mutex
	w  io.Writer

	errorf func(f string, v ...interface{})
}
```

File: cli/support.go [hunk 5]
```diff
@@ -48,14 +51,52 @@ func (r *RootCmd) supportBundle() *serpent.Command {
...
+					Default:   cliui.ConfirmNo,
+					Secret:    false,
+					IsConfirm: true,
+				})
+				if err != nil || ans != cliui.ConfirmYes {
+					return err
+				}
+				cliLog.Info(inv.Context(), "user confirmed manually", slog.F("answer", ans))
+			} else {
+				cliLog.Info(inv.Context(), "user auto-confirmed")
```
Comment (mtojek) [inline cli/support.go:84]: nit: debug or trace level?

File: cli/support.go [hunk 6]
```diff
@@ -48,14 +51,52 @@ func (r *RootCmd) supportBundle() *serpent.Command {
...
+			}
+
+			// Check if we're running inside a workspace
+			if val, found := os.LookupEnv("CODER"); found && val == "true" {
```
Comment (mtojek) [inline cli/support.go:88]: nit: maybe create a util function for this?

File: cli/support.go [hunk 7]
```diff
@@ -110,12 +169,24 @@ func (r *RootCmd) supportBundle() *serpent.Command {
 	}
 	cmd.Options = serpent.OptionSet{
 		{
-			Flag:          "output",
-			FlagShorthand: "o",
-			Env:           "CODER_SUPPORT_BUNDLE_OUTPUT",
+			Flag:        "confirm",
```
Comment (mtojek) [inline cli/support.go:172]: nit: `--yes`  or `--silent`?

Comment (johnstcn) [reply cli/support.go:189]: Forgot about `cliui.SkipPromptOption`

File: cli/support.go (hunk 5 above)
Comment (johnstcn) [reply cli/support.go:101]: Yeah fair, we don't want unnecessary output.

**Pull request merged**